go/vt/vitessdriver/driver.go

```diff
@@ -253,103 +214,105 @@ func (c *conn) Begin() (driver.Tx, error) {
 }
 
 func (c *conn) Commit() error {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
```

Comment:
I see vitess takes a context with cancellation for Commit / Rollback(). Would it be good for the SQL driver to do the same?